app/src/main/java/com/github/polypoly/app/network/storable/StorableObject.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.polypoly.app.network.storable
+
+import kotlin.reflect.KClass
+
+/**
+ * [StorableObject] is the super class of all classes that'll be contained in the DB.
+ * It implements all the abstractions needed by the remote storage to do all the queries, i.e.
+ * - The [dbPath] that precedes every key for a given class (e.g. "objectX/"),
+ * - The converter (to local) that converts a T DB object to a local StorableObject<T>,
+ * - The converter (to DB)]that converts an instance of a local StorableObject<T> to a DB T.
+ *
+ * It is the role of every subclass to define the path and the conversion methods.
+ *
+ * This class takes a generic type T as argument, corresponding to the data class that is
+ * going to be stored in the DB.
+ *
+ * @note attribute [key] should be the primary key used to store the object in the DB
+ *
+ * @note for the subclass to be correctly initialized, one instance should be created before any operation
+ */
+abstract class StorableObject<T : Any> (dbPath: String, private val key: String) {
```

Comment:
should `dbPath` be private as well?

Reply:
About this, shouldn't the key be the path as well?

Reply:
Or it would make more sense to make the path private but the key public?

Reply:
`dbPath` has no `val` associated to it so no one can access `dbPath`, so it is already a way of private.
The `key` is however meant to be public and was private for immutability reasons as mentioned below but now it is fixed :)